src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/AuditEvent.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model
+
+import com.fasterxml.jackson.annotation.JsonFormat
+import io.swagger.v3.oas.annotations.media.Schema
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.AuditEventType
+import java.time.LocalDateTime
+
+@Schema(description = "Describes an audit event request")
+data class AuditEvent(
+  @Schema(description = "The internal ID of the audit event", example = "1234")
+  val id: Long? = null,
```

Comment:
I don't generally like using the same DTO for request and response. The request to create the audit would not have an ID supplied, and may actually cause error if the ID _was_ supplied. Whereas the response will have an ID returned. The request does not look the same as the response and I would usually split the DTOs into request DTO and response DTO.

Although, this is a wider issue with the API, not just related to this

Reply:
The same object is used as its identical (almost!). The ID is nullable for requests, so made sense for convenience, and keeping it simple. This might evolve - I'm already wondering about whether I should record specific event_types e.g. login, logout, create, approve, vary, edit, print, view etc.. Might be useful to be able to query all events for create etc.  Could also include the user type (prison or probation), and which prison or team ... but for now, I thought I'd do the minimum, and we can evolve the needs arise.